src/crypto/algorithms/megolm.ts

```diff
@@ -592,7 +592,7 @@ class MegolmEncryption extends EncryptionAlgorithm {
         payload: IPayload,
```

Comment:
> Sonar now ants 80% coverage on this new code, while I'm not sure how to achieve that. Any suggestions? Adding specific tests for this?

It would be good to add a `encryptAndSendKeysToDevices` test which hits on this edge case you're trying to solve:

> This assumes that no two users share a device with the same deviceID, though that is possible given a large enough room and a large enough time for users to keep generating deviceIDs.
>
> *-- https://github.com/matrix-org/matrix-js-sdk/issues/2135#issue-1115248936*

Reply:
Sonar seems to be okay with the new lines, i had this problem before in another PR, i merged `develop` into here again to use the new rules.